src/main/java/Duke.java

```diff
@@ -1,10 +1,208 @@
+//import java.text.ParseException;
+
 public class Duke {
-    public static void main(String[] args) {
+
+    private ArrayList<Task> list;
+    private Saved savedFile;
+    private Scanner scan;
+    final private static String LINE = "____________________________________________________________";
+
+    public static void main(String[] args) throws IOException, ParseException {
+        new Duke("src/main/java/data.txt").run();
+    }
+
+    public Duke(String filePath) throws IOException, ParseException {
+        scan = new Scanner(System.in);
+        savedFile = new Saved(filePath);
+        list = savedFile.loadData();
+    }
+
+    public void run() throws IOException, ParseException {
+
+        final String GOT_IT = "Got it. I've added this task:";
+        String cmd;
+
+        printIntro();
+
+        Boolean isNotBye = true;
+        while(isNotBye) {
+            cmd = scan.nextLine();
+            String cmdArr[] = cmd.split(" ", 2);
+            String firstWord = cmdArr[0];
+
+            switch (firstWord) {
+            case "bye":
+                printLine();
+                printDuke("Bye. Hope to see you again soon!");
+                printLine();
+                isNotBye = false;
+                break;
+
+            case "list":
+                printLine();
+                if (list.isEmpty()) {
+                    System.out.println("    There is nothing in your list.");
+                    printLine();
+                    break;
+                }
+                printList(list);
+                break;
+
+            case "done":
+                printLine();
+                int taskNum = Integer.valueOf(cmdArr[1]);
+                list.get(taskNum - 1).isDone(true);
+                printDuke("Nice! I've marked this task as done");
+                System.out.println("    " + taskNum + ". [" + list.get(taskNum - 1).getStatusIcon() + "]"
+                                        + list.get(taskNum - 1).getDesc());
+                printLine();
+                break;
+
+            case "todo":
+                try {
+                    String task = cmdArr[1];
+                    Todo todo = new Todo(task);
+                    list.add(todo);
+
+                    //output
+                    printLine();
+                    printDuke(GOT_IT);
+                    printDuke(todo.toString());
+                    printDuke("Now you have " + list.size() + " tasks in the list");
+                    printLine();
+                } catch (ArrayIndexOutOfBoundsException exception) {
+                    printLine();
+                    printDuke("\u2639 OOPS!!! The description of a todo cannot be empty.");
+                    printLine();
+                }
+                break;
+
+            case "deadline":
+                try {
+                    String taskAndDate = cmdArr[1];
+                    String dl[] = taskAndDate.split("/", 2);
+                    try {
```

Comment:
Abstract this `try ... catch ...` block into a separate method to avoid nested `try ... catch ...` .